### ISSUE TITLE
Prefer dataObjectOf functions instead of the DataObject ones

### ISSUE THREAD

**Issue by gmariotti** [vert-x3/vertx-lang-kotlin#103]
This proposal is to change the current dataobject helper functions from `fun DataObject(..): io.vertx.DataObject` to `fun dataObject(..): io.vertx.DataObject`.

The reasons for this change are:
- Avoid collisions when both the dataobject and the helper function needs to be imported in the same file
```kt
// Doesn't compile
suspend fun sendRequest(client: WebClient): JsonObject { // imported from io.vertx.core.json
  val request = JsonObject("key" to "value") // imported from io.vertx.kotlin.core.json
  return webClient.getAbs("path").sendAwait().bodyAsJsonObject()
}
```
- align with functions like `mapOf` and `listOf` available in `kotlin-stdlib`

**Comment by gmariotti** [comment]
Agree, probably adding `of` at the end will make it more explicit, let's see if anybody else is in the community is interested in it.

**Comment by inikolaev** [comment]
+1 for the `Of` suffix

**Comment by gmariotti** [comment]
Thinking about it, this change doesn't need to be a breaking one. As I wrote in #110, I think we can generate both methods and deprecate the old one with `@Deprecated(replaceWith = ReplaceWith("newMethod"))` and officially remove the methods in Vert.x 4.1. Thoughts about it?

**Comment by vietj** [comment]
@gmariotti agreed it would be the way to go for 3.x

**Comment by gmariotti** [comment]
@vietj are you expecting any 3.7 version? Or do you want to deprecate the old methods already in 3.6?

**Comment by vietj** [comment]
we don't foresee yet a 3.7 (but it's not excluded). so 3.6 is fine to me since we are also deprecating other v4 things.

**Comment by gmariotti** [comment]
Closed by #110, the next step is to backport it to 3.6 or 3.7

**Comment by gmariotti** [comment]
Backported to the 3.6 branch, it will be available starting from 3.6.3
